staticaddr/loopin/actions.go

```diff
@@ -0,0 +1,594 @@
+	err := f.cfg.DepositManager.TransitionDeposits(
+		f.loopIn.Deposits, deposit.OnLoopedIn, deposit.LoopedIn,
+	)
+	if err != nil {
+		return f.HandleError(err)
+	}
+
+	return OnSignSweeplessSweep
+}
+
+// SignSweeplessSweepAction is called if the invoice was settled. We sign the
+// servers request to sweep the deposits to the server's sweep address.
+func (f *FSM) SignSweeplessSweepAction(_ fsm.EventContext) fsm.EventType {
+	// Create a musig2 session for each deposit.
+	musig2Sessions, clientNonces, err := f.loopIn.createMusig2Sessions(
+		f.ctx, f.cfg.Signer,
+	)
+	if err != nil {
+		return f.HandleError(err)
+	}
+	f.sweeplessMusig2Sessions = musig2Sessions
+	f.sweeplessClientNonces, err = toNonces(clientNonces)
+	if err != nil {
+		return f.HandleError(err)
+	}
+
+	// Create the sweepless sweep transaction.
+	sweeplessSweepTx, err := f.loopIn.createSweeplessSweepTx()
+	if err != nil {
+		return f.HandleError(err)
+	}
+
+	// Next we'll get our htlc tx signatures.
+	sweeplessSigs, err := f.loopIn.signMusig2Tx(
+		f.ctx, sweeplessSweepTx, f.cfg.Signer,
+		f.sweeplessMusig2Sessions, f.sweeplessServerNonces,
+	)
+	if err != nil {
+		return f.HandleError(err)
+	}
+	f.sweeplessClientSigs = sweeplessSigs
+
+	// Push Htlc sigs to server
+	pushSweeplessSigsReq := &looprpc.PushStaticAddressSweeplessSigsRequest{ //nolint:lll
+		SwapHash:              f.loopIn.SwapHash[:],
+		SweeplessClientNonces: fromNonces(f.sweeplessClientNonces),
+		SweeplessClientSigs:   f.sweeplessClientSigs,
+	}
+	_, err = f.cfg.StaticAddressServerClient.PushStaticAddressSweeplessSigs( //nolint:lll
+		f.ctx, pushSweeplessSigsReq,
+	)
+	if err != nil {
+		return f.HandleError(err)
+	}
+
+	return OnSweeplessSweepSigned
+}
+
+// UnlockDepositsAction is called if the loop-in failed and its deposits should
+// be available in a future loop-in request.
+func (f *FSM) UnlockDepositsAction(_ fsm.EventContext) fsm.EventType {
+	err := f.cfg.DepositManager.TransitionDeposits(
+		f.loopIn.Deposits, fsm.OnError, deposit.Deposited,
+	)
+	if err != nil {
+		return f.HandleError(err)
+	}
+
+	return fsm.OnError
+}
+
+// createAndPublishHtlcTimeoutSweepTx creates and publishes the htlc timeout
+// sweep transaction.
+func (f *FSM) createAndPublishHtlcTimeoutSweepTx() error {
+	// Get a fee rate.
+	feeRate, err := f.cfg.WalletKit.EstimateFeeRate(
+		f.ctx, defaultConfTarget,
+	)
+	if err != nil {
+		return err
+	}
+
+	getInfo, err := f.cfg.LndClient.GetInfo(f.ctx)
+	if err != nil {
+		return err
+	}
+
+	// Create htlc timeout transaction.
+	timeoutTx, err := f.loopIn.createHtlcSweepTx(
+		f.ctx, f.cfg.Signer, f.loopIn.HtlcTimeoutSweepAddress, feeRate,
+		f.cfg.ChainParams, getInfo.BlockHeight,
+	)
+	if err != nil {
+		return err
+	}
+
+	// Broadcast htlc timeout transaction.
+	txLabel := fmt.Sprintf("htlc-timeout-sweep-%v",
+		f.loopIn.SwapHash)
+
+	err = f.cfg.WalletKit.PublishTransaction(f.ctx, timeoutTx, txLabel)
+	if err != nil {
+		if !strings.Contains(err.Error(), "output already spent") {
```

Comment:
I think there may be a few more error cases to handle, probably `chain.ErrInsufficientFee` being the most important.